client/findproviders.go

```diff
@@ -1,135 +1,116 @@
 package client
 
 import (
-	"bytes"
 	"context"
-	"encoding/json"
-	"errors"
-	"io"
-	"net/http"
-	"net/url"
 
 	"github.com/ipfs/go-cid"
-	"github.com/ipfs/go-delegated-routing/parser"
+	proto "github.com/ipfs/go-delegated-routing/gen/proto"
 	logging "github.com/ipfs/go-log"
 	"github.com/libp2p/go-libp2p-core/peer"
 	"github.com/multiformats/go-multiaddr"
 )
 
-var logger = logging.Logger("delegated/client")
+var logger = logging.Logger("service/client/DelegatedRouting")
 
-func (c *client) FindProviders(ctx context.Context, cid cid.Cid) ([]peer.AddrInfo, error) {
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
-	ch, err := c.FindProvidersAsync(ctx, cid)
+type Client struct {
+	client proto.DelegatedRouting_Client
+}
+
+func NewClient(c proto.DelegatedRouting_Client) *Client {
```

Comment:
Let's add another wrapper package with some code getting this to implement the relevant libp2p-core routing interfaces (e.g. https://github.com/libp2p/go-libp2p-core/blob/374bb62823579bd73a5e2c6cfaf19de24f6cbd5a/routing/routing.go#L26 and https://github.com/libp2p/go-libp2p-core/blob/374bb62823579bd73a5e2c6cfaf19de24f6cbd5a/routing/routing.go#L49).

This will make it easier for people to use tools like https://github.com/libp2p/go-libp2p-routing-helpers/blob/506670d53a31503f257980dbf354b3ea1b990fb7/parallel.go#L20

Reply:
I've added a wrapper for the ContentRouting interface.

Are you sure you want a wrapper for the ValueStore interface? This interface doesn't seem to map cleanly to the FindProviders/GetIPNS/PutIPNS interface. The Get path would require trying to decode the stored record either as a provider or as an IPNS record. Is there a use case for this?

Also, let's not extend the scope of this PR beyond what is necessary for Hydra and the Delegated Routing code path in IPFS.

Reply:
Makes sense. Let me open a new issue about adding the `ValueStore` interface. I want to unblock the integration with hydra for now, without expanding scope since it's been too long this issue has been open. Added https://github.com/ipfs/go-delegated-routing/issues/13